src/wallet/rpc/addresses.cpp

```diff
@@ -578,7 +578,7 @@ RPCHelpMan getaddressinfo()
 
     if (provider) {
         auto inferred = InferDescriptor(scriptPubKey, *provider);
-        bool solvable = inferred->IsSolvable() || IsSolvable(*provider, scriptPubKey);
+        bool solvable = inferred->IsSolvable();
```

Comment:
nit: 
```suggestion
        bool solvable = inferred && inferred->IsSolvable();
```

Reply:
I don't want to change the behaviour in this pull. The current code is an implicit assertion that we can infer a descriptor if we could get a provider (we'd crash otherwise), and this patch would silently mark the address as not solvable instead.

Reply:
In that sense, then might be better to add an explicit assertion so we don't rely on a nullptr crash?
```c++
auto inferred = Assert(InferDescriptor(scriptPubKey, *provider));
```
(np on doing it here or not)

Reply:
I added an explicit assertion, but there is a linter failure about using `CHECK_NONFATAL` instead because it happens to be in RPC code.. But it's not nonfatal, currently it is. Anyhow, it's not related to this PR, so i just pushed back the previous state of this branch without the assertion.